### ISSUE TITLE
Applying brain-age pretrained models to external cohorts

### ISSUE THREAD

**Issue by jhuguetn** [estenhl/pyment-public#30]
Hi @estenhl,

First, thank you for sharing `pyment` and its pretrained models, this is a very valuable contribution to the community!  
 
I am currently evaluating your models on my own data and plan to systematically extract brain age metric as a imaging biomarker in dementia/AD related analyses.

As part of my validation, I tested the SFCN-sm (softmax), the pretrained model initially showing more promising results,  on 2 cohorts:
* Cohort-1: ~160 mostly healthy individuals, ages 50–70
* Cohort-2: ~120 individuals, ages 60–90, including SCU and MCI cases (no AD pathology)

My preprocessing stage followed your provided [freesurfer_and_fsl.sh](https://github.com/estenhl/pyment-public/blob/main/preprocessing/freesurfer_and_fsl.sh) workflow, though with different software versions (FS7.1 and FSL 6.0.7). For simplicity, I re-used existing FreeSurfer recon-all (full pipeline) outputs, running only the MNI registration and cropping preproc. steps. 
One detail worth mentioning: my T1w MRI data has high resolution (0.75 mm isotropic). I retained this native resolution by disabling FreeSurfer’s default downsampling to 1 mm, so all preprocessing was carried out on the higher-resolution data.  

In both cohorts, the uncorrected MAE is ~9 years, which is much higher than the value reported in the paper for SFCN-reg (and also above the ~5 MAE reported for softmax in this repo). Age-stratified MAE shows especially poor performance in age ranges underrepresented in your training distribution (40–55 and ≥75). See below,

<img width="500" height="250" alt="Image" src="https://github.com/user-attachments/assets/30592202-68ae-4680-808d-e916600f07c4" />

<img width="500" height="250" alt="Image" src="https://github.com/user-attachments/assets/9f134402-dcb5-4f1f-bf35-80ab4b33b1b2" />

My questions are:
- Is this level of performance expected when applying the models to external data? 
- Do you recommend applying bias correction (or modelling chronological age as a covariate) when using pyment predictions outside the training cohorts? 
- Are there preprocessing details that you have found to be particularly critical for model accuracy? 
  
Any advice/suggestion would be greatly appreciated. Thanks again for making this work available, it is very helpful.

Jordi

**Comment by jhuguetn** [comment]
Hi @estenhl! No worries at all, and thanks for following up.

I think we now have a reasonably robust setup and workflow around the pretrained SFCN-sm model. What I finally decided to do was to systematically run the brain-age prediction on all our MRI experiments using your pretrained model and store the raw predictions. Then, depending on the downstream analysis (cross-sectional, longitudinal, multi-cohort, etc...), we perform bias correction on-the-fly at the cohort/dataset level rather than globally.
Predicted ages are corrected by residualizing raw predicted age against chronological age according to the specific population distribution, following previously described brain-age bias correction approaches (Peng et al., 2021). The corrected BAG is then calculated as the difference between bias-corrected predicted age and chronological age.

The uncorrected MAE on our external cohorts remained higher than reported in the original paper (~9 years before correction), although bias correction reduced this substantially (~4.5 years after linear correction). We suspect part of the discrepancy may relate to age distribution differences and underrepresentation of some age ranges in the original training data, but overall the framework behaves consistently enough for our downstream analyses.

**Comment by estenhl** [comment]
Glad to hear, let me know if there's anything else!
